Lib/python/director.swg

```diff
@@ -269,6 +269,30 @@ namespace Swig {
     static void raise(const char *msg) {
       throw DirectorMethodException(msg);
     }
+
+    /* this should be called only if a Python error has occurred */
+    static void raise_after_pyerr(const char *msg) {
+      std::string full_msg(msg);
+
+      PyObject *exception, *v, *tb;
+      PyErr_Fetch(&exception, &v, &tb);
+      PyErr_NormalizeException(&exception, &v, &tb);
+
+      if (PyObject *s = PyObject_Str(v)) {
+        if (const char *cstr = SWIG_Python_str_AsChar(s)) {
```

Comment:
How likely are these checks to fail?  We could avoid converting `msg` to `std::string` and back when we're just going to use it unaltered.

Reply:
AFAIK they can never fail at all, the checks are just there to account for the "impossible" failures.

Reply:
OK, then there's no point complicating things.

It'd be good to have a testcase covering this change, but otherwise it looks sensible to me.